### DISCUSSION TITLE
nested model

### DISCUSSION THREAD

**Discussion by matv864** [igorbenav/fastcrud#90] [Q&A]
I try to create api on sqlmodel with relationships (In my case, I have table Card and Article with link to Card (key: card_id))
one-to-many

And now I try to understand how it works (joined isn't my solving, I need record which contain Card model with list of Article models inside)
![image](https://github.com/igorbenav/fastcrud/assets/91239322/28997837-aa82-43b2-aeea-e054894fb208)

code of getting record

```python
return await card_crud.get(
                db=db,
                id=card_id,
                schema_to_select=Card_schema,
            )
```

result of this response
![image](https://github.com/igorbenav/fastcrud/assets/91239322/c233221b-066a-4ed1-8a9b-c0ccaededc31)

Why  "true"   ....?
How did it appeared ....?

how can I implement this?

**Comment by matv864** [comment]
in Usual SQL I would use this
[selectinload](https://docs.sqlalchemy.org/en/20/orm/queryguide/relationships.html#sqlalchemy.orm.selectinload)

but how can I make this in FastCRUD

**Comment by igorbenav** [comment]
I think what you are looking for is the [get_joined](https://igorbenav.github.io/fastcrud/usage/crud/#2-get-joined) method. 

Here's a [comprehensive guide](https://igorbenav.github.io/fastcrud/advanced/joins/) to joins in `FastCRUD`, including nested joins

**Comment by matv864** [comment]
I don't found this article when read documentation, and  it is my case.

But how can I make nested many records?

![image](https://github.com/igorbenav/fastcrud/assets/91239322/2225d7c1-ad20-4e2d-b751-5d6368439ec4)

![image](https://github.com/igorbenav/fastcrud/assets/91239322/33ce2b21-a019-470f-9030-b014807716e9)

```python
async def get_card(self, card_id: uuid.UUID):
        async with async_session_maker() as db:
            return await card_crud.get_joined(
                db=db,
                id=card_id,
                nest_joins=True,
                joins_config=[
                    JoinConfig(
                        model=Article,
                        join_on=Article.card_id == Card.id,
                        join_type="left",
                        join_prefix="articles_",
                        schema_to_select=Article_schema,
                    )
                ]
            )
```

**Reply by igorbenav** [reply]
Now that I actually understood, this apparently is a bug indeed. `Articles` in the response body should be a list. I'll fix this

**Reply by iameo** [reply]
I started using Fastcrud just last; loving it thus far, @igorbenav. However, I have had this issue, too, for at least a day. Is it okay if I asked for the progress to ensuring multiple entries, in this case, Article, are in a list? Thank you for the gift that is Fastcrud. 

I plan on going through the source code later today to see if I can figure the issue out.

**Reply by igorbenav** [reply]
Hey, @iameo, I'm glad you like it!
I'll try to fix it today or tomorrow (I just need to finish #87 docs and tests), 0.13.0 with the fix should be released by Wednesday or Thursday.

**Reply by igorbenav** [reply]
Related issue here #91. It will take longer than I initially thought, it isn't that simple. Release with the fix will probably be done by monday

**Comment by igorbenav** [comment]
The pr #93 will fix the related issue in #91. I just need to write the docs.
